pkg/kubelet/remote/remote_runtime.go

```diff
@@ -151,7 +155,9 @@ func (r *RemoteRuntimeService) CreateContainer(podSandBoxID string, config *runt
 	ctx, cancel := getContextWithTimeout(r.timeout)
 	defer cancel()
 
+	uid := uuid.NewUUID()
```

Comment:
Using the pod uid for the pod sandbox create makes sense. We should do that and I'm waiting on the other pr for that.

For the container create it's much harder to pick a good value. This is definitely the worst possible choice and I figured this discussion would happen. That being said, this still provides some minimum protection (network errors/retries within the client for a single request).

If this is something that grpc already magically does on a per-request level, then this level of implementation is useless, but I don't think it does.


Other things we can try to protect against are the following:

1) Retries within the kubelet package - I don't think the kubelet is the right layer to retry at, but rather the runtime, so I don't think this is something to protect against. To do that, the kubelet would need to pass in a uid rather than this runtime client.
2) Linearalizability errors between "List()" and "Create()" calls - This is something that cannot be done by a dumb idempotency token, but I also think we should assume our runtime will return a given container in list post-create.
3) kubelet crashes, or timeouts such that 2 is not true (where such issues can be expected).

0, network/grpc related retries and 3, kubelet crashes / call timeouts, are the ones I see value in making idempotent.

Unfortunately, picking a good container value is hard because multiple instances of a given container name in a pod may exist at the same time (container + new copy due to restart policy), so PodUID  + name isn't enough.

I'll switch it to assuming we have a tuple of (ID, name, instanceNumber) available, but I wouldn't be surprised if generating that number (presumably pick the largest number you see in list) has its own host of problems.